### PR TITLE
refactor: GridForm-test

### PR DESCRIPTION
**Pull request by dreamwasp**
### PR Checklist
Refactors `Gridform-tests` and some other small improvements 

- [ ] Related to designs:
- [x] Related to JIRA ticket: [GM-289](https://codecademy.atlassian.net/browse/GM-289)
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.69-alpha.67870b.0<br/>@codecademy/gamut-labs@19.16.7-alpha.67870b.0<br/>@codecademy/gamut@35.0.2-alpha.67870b.0<br/>@codecademy/styleguide@42.0.2-alpha.67870b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f742e60ce5291bb3d39beb--gamut-preview.netlify.app>https://60f742e60ce5291bb3d39beb--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f742e60ce5291bb3d39beb>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.69-alpha.dd6555.0<br/>@codecademy/gamut-labs@19.16.7-alpha.dd6555.0<br/>@codecademy/gamut@35.0.2-alpha.dd6555.0<br/>@codecademy/styleguide@42.0.2-alpha.dd6555.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f747db1c112b205fca1011--gamut-preview.netlify.app>https://60f747db1c112b205fca1011--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f747db1c112b205fca1011>Deploy Logs</a></p>

**Review by codecaaron — APPROVED** [review]

File: packages/gamut/src/GridForm/__tests__/GridForm-test.tsx [hunk 1]
```diff
@@ -19,44 +17,63 @@ import {
...
 
+const fields = [stubCheckboxField, stubSelectField, stubTextField];
+
+const renderView = setupRtl(GridForm, {
+  fields,
+  showRequired: false,
+  submit: { type: 'fill', contents: <>Submit</>, size: 6 },
+});
+
+type renderViewReturn = ReturnType<typeof renderView>;
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/GridForm/__tests__/GridForm-test.tsx:28]: Nit: this is generally done:

```suggestion
type RenderViewReturn = ReturnType<typeof renderView>;
```

File: packages/gamut/src/GridForm/__tests__/GridForm-test.tsx [hunk 2]
```diff
@@ -19,44 +17,63 @@ import {
...
+
+const asyncRenderView = async (
+  props: Partial<
+    GridFormProps<Record<string, string | boolean | FileList | undefined>>
+  >
+): Promise<renderViewReturn> => {
+  let renderResults: renderViewReturn;
+
+  await act(async () => {
+    renderResults = await renderView(props);
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/GridForm/__tests__/GridForm-test.tsx:38]: Nit: a little less hardcode-y

```suggestion
const asyncRenderView = async (
  ...props: Parameters<typeof renderView>
): Promise<renderViewReturn> => {
  let renderResults: renderViewReturn;

  await act(async () => {
    renderResults = await renderView(...props);
```

File: packages/gamut/src/GridForm/__tests__/GridForm-test.tsx [hunk 3]
```diff
@@ -82,268 +97,195 @@ describe('GridForm', () => {
...
     await act(async () => {
-      fireEvent.click(screen.getByRole('button'));
+      fireEvent.click(view.getByRole('button'));
     });
 
     // there should only be a single "assertive" error from the form submission
-    expect(await screen.queryAllByRole('alert').length).toBe(1);
-    expect(await screen.queryAllByRole('status').length).toBe(1);
+    expect(await view.queryAllByRole('alert').length).toBe(1);
+    expect(await view.queryAllByRole('status').length).toBe(1);
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/GridForm/__tests__/GridForm-test.tsx:108]: Nit: while you're here, mind fixing these? I just noticed these don't need to `await`...

```suggestion
    expect(view.getAllByRole('alert').length).toBe(1);
    expect(view.getAllByRole('status').length).toBe(1);
```

File: packages/gamut/src/GridForm/__tests__/GridForm-test.tsx [hunk 4]
```diff
@@ -82,268 +97,195 @@ describe('GridForm', () => {
...
-
-    expect(form.find('input#mycoolid').length).toBe(1);
-    expect(form.find('select#swaggy-id').length).toBe(1);
-    expect(form.find('input#another-dank-id').length).toBe(1);
-    expect(form.find('input#name-0-and-another-one').length).toBe(1);
-    expect(form.find('input#name-1-and-another-one').length).toBe(1);
-    expect(form.find('input#another-dank-id').length).toBe(1);
-    expect(form.find('textarea#id-2-the-ego').length).toBe(1);
-    expect(form.find('input#fire-file').length).toBe(1);
+  it('passes custom ids to the fields', async () => {
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/GridForm/__tests__/GridForm-test.tsx:225]: Just noticing this now: feels like this test should really be a `.each`. A single one of these failing (e.g. `textField`'s `mycoolid`) would prevent us from seeing results on later ones.

**Review by JoshuaKGoldberg — APPROVED** [review]
Cool!

Some nits, mostly on existing code.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.69-alpha.db247f.0<br/>@codecademy/gamut-labs@19.16.7-alpha.db247f.0<br/>@codecademy/gamut@35.0.2-alpha.db247f.0<br/>@codecademy/styleguide@42.0.2-alpha.db247f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f8727a1c49381f626c8be6--gamut-preview.netlify.app>https://60f8727a1c49381f626c8be6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f8727a1c49381f626c8be6>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.69-alpha.9bae09.0<br/>@codecademy/gamut-labs@19.16.7-alpha.9bae09.0<br/>@codecademy/gamut@35.0.2-alpha.9bae09.0<br/>@codecademy/styleguide@42.0.2-alpha.9bae09.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f873e951a0cf23bb219339--gamut-preview.netlify.app>https://60f873e951a0cf23bb219339--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f873e951a0cf23bb219339>Deploy Logs</a></p>

**Pull request merged**